### PR TITLE
Sploit 947 | Added EC2 Open Custom Ports plugin, spec file

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/aws/ec2/openCustomPorts.spec.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                        "CidrIp": "0.0.0.0/0"
+                    }
+                ],
+                "Ipv6Ranges": [],
+                "PrefixListIds": [],
+                "ToPort": 80,
+                "UserIdGroupPairs": []
+            }
+        ],
+        "OwnerId": "560213429563",
```
Comment (giorod3) [inline plugins/aws/ec2/openCustomPorts.spec.js:66]: is that your account id?

Comment (AkhtarAmir) [reply plugins/aws/ec2/openCustomPorts.spec.js:66]: yeah, let me update it.

File: plugins/aws/ec2/openCustomPorts.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    more_info: 'Security groups should be used to restrict access to ports from known networks.',
+    link: 'https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html',
+    recommended_action: 'Modify the security group to ensure the ports are not exposed publicly.',
+    apis: ['EC2:describeSecurityGroups', 'STS:getCallerIdentity'],
+    settings: {
+        open_port_allowed_list: {
+            name: 'EC2 Allowed Open Ports',
+            description: 'A comma-delimited list of ports that indicates open ports allowed for any connection',
+            regex: '[a-zA-Z0-9,]',
+            default: [80, 443]
```
Comment (giorod3) [inline plugins/aws/ec2/openCustomPorts.js:17]: this should be a string not an array. Example here: https://github.com/aquasecurity/cloudsploit/blob/master/plugins/aws/cloudformation/plainTextParameters.js#L17

Comment (AkhtarAmir) [reply plugins/aws/ec2/openCustomPorts.js:17]: Like this: '80,443'?

Comment (giorod3) [reply plugins/aws/ec2/openCustomPorts.js:17]: Yes but you will have to refactor the code to split the string into an array.

**Comment by AkhtarAmir** [comment]
got it

**Review by giorod3 — APPROVED** [review]

**Review by giorod3 — APPROVED** [review]

File: plugins/aws/ec2/openCustomPorts.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                    }
+                }
+
+                if (!cidrIps.length) {
+                    helpers.addResult(results, 0,
+                        `Security group "${group.GroupName}" does not have ports open to 0.0.0.0/0`, 
+                        region, resource);
+                } else {
+                    helpers.addResult(results, 2,
+                        `Security group "${group.GroupName}" has ports open to ${cidrIps}`,
```
Comment (AkhtarAmir) [inline plugins/aws/ec2/openCustomPorts.js:118]:
```suggestion
                        `Security group "${group.GroupName}" has ports open to ${cidrIps.join(', ')}`,
```

**Comment by giorod3** [comment]
merged to pr bundle

**Comment by giorod3** [comment]
akhtar making changes in bundle

**Pull request closed without merging**